leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/attributes/WriteAttributeObserveTest.java

```diff
@@ -368,6 +451,214 @@ public void test_with_negative_gap_on_unsigned_integer_resource(Protocol givenPr
+
+        // Changing value which crosses the lt limit
+        writeResponse = server.send(currentRegistration, new WriteRequest(TEST_OBJECT, 0, INTEGER_VALUE, 400));
+        assertThat(writeResponse).isSuccess();
+
+        // Verify Behavior
+        ObserveResponse response = server.waitForNotificationOf(observation,
+                (int) (TimeUnit.MILLISECONDS.convert(pmax, TimeUnit.SECONDS) * 0.3), TimeUnit.MILLISECONDS);
+        assertThat(response.getContent()).isEqualTo(LwM2mSingleResource.newIntegerResource(INTEGER_VALUE, 400));
+        assertThat(response).hasValidUnderlyingResponseFor(givenServerEndpointProvider);
+        server.ensureNoNotification(observation, 500, TimeUnit.MILLISECONDS);
+    }
+
+    @TestAllTransportLayer
+    public void test_pmin_equals_pmax(Protocol givenProtocol, String givenClientEndpointProvider,
+                                      String givenServerEndpointProvider) throws InterruptedException {
+
+        Long pmin = 1l; // seconds
+        Long pmax = 1l; // seconds
+
+        // Setting attribute
+        WriteAttributesResponse writeAttributeResponse = server.send(currentRegistration,
+                new WriteAttributesRequest(TEST_OBJECT, 0, INTEGER_VALUE,
+                        new LwM2mAttributeSet(LwM2mAttributes.create(LwM2mAttributes.MINIMUM_PERIOD, pmin),
+                                LwM2mAttributes.create(LwM2mAttributes.MAXIMUM_PERIOD, pmax))));
+
+        // Verify write attribute response
+        assertThat(writeAttributeResponse).isSuccess();
+
+        // Setting observe
+        ObserveResponse observeResponse = server.send(currentRegistration,
+                new ObserveRequest(TEST_OBJECT, 0, INTEGER_VALUE));
+        assertThat(observeResponse).isSuccess();
+
+        SingleObservation observation = observeResponse.getObservation();
+        assertThat(observation.getPath()).isEqualTo(new LwM2mPath(TEST_OBJECT, 0, INTEGER_VALUE));
+        server.waitForNewObservation(observation);
+
+        // Trigger new notification
+        LwM2mResponse writeResponse = server.send(currentRegistration,
+                new WriteRequest(TEST_OBJECT, 0, INTEGER_VALUE, 30));
+        assertThat(writeResponse).isSuccess();
+
+        // Verify
+        server.ensureNoNotification(observation, (int) (TimeUnit.MILLISECONDS.convert(pmin, TimeUnit.SECONDS) * 0.8),
+                TimeUnit.MILLISECONDS);
+        ObserveResponse response = server.waitForNotificationOf(observation,
+                (int) (TimeUnit.MILLISECONDS.convert(pmin, TimeUnit.SECONDS) * 0.3), TimeUnit.MILLISECONDS);
+        assertThat(response.getContent()).isEqualTo(LwM2mSingleResource.newIntegerResource(INTEGER_VALUE, 30));
+        assertThat(response).hasValidUnderlyingResponseFor(givenServerEndpointProvider);
+    }
+
+    @TestAllTransportLayer
+    public void test_objectId_inheritance(Protocol givenProtocol, String givenClientEndpointProvider,
```

Comment:
Ok so tests didn't pass because there was a regression in the code. (I fixed it and pushed it in `write_attribute` branch.

Reading more about all of those, I think that the specification doesn't allow to attach `lt`, `gt` or `st` to Object level. 
(In LWM2M v1.1.1 you need to look at assignation level column, see https://github.com/eclipse-leshan/leshan/issues/1588)

So you should do your inheritance test using pmin or pmax (maybe better to use pmax so you don't need to send a write request)

Sorry if all of that is confusing but write attribute is not really an easy feature. :grimacing:

Reply:
I changed and add more tests about inheritance using pmax (as supported by all level and so code can be factorized).